### ISSUE TITLE
bad argument in arithmetic expression :ldclient_backoff.backoff/2

### ISSUE THREAD

**Issue by zbarnes757** [launchdarkly/erlang-server-sdk#102]
There seems to be an arithmetic error in the ldclient backoff module. All variables seem to be set by the module so I am unsure where the problem is coming from.

**To reproduce**

All this happens within the ldclient. There is nothing done on my end to trigger this issue.

**Expected behavior**

This error should be handled by coercing to the proper type.

**Logs**

```elixir
GenServer #PID<0.91205.0> terminating
** (ArithmeticError) bad argument in arithmetic expression
    (ldclient 2.1.2) /opt/app/deps/ldclient/src/ldclient_backoff.erl:98: :ldclient_backoff.backoff/2
    (ldclient 2.1.2) /opt/app/deps/ldclient/src/ldclient_backoff.erl:94: :ldclient_backoff.delay/2
    (ldclient 2.1.2) /opt/app/deps/ldclient/src/ldclient_backoff.erl:90: :ldclient_backoff.update_backoff/2
    (ldclient 2.1.2) /opt/app/deps/ldclient/src/ldclient_update_stream_server.erl:93: :ldclient_update_stream_server.handle_info/2
    (stdlib 5.0.2) gen_server.erl:1077: :gen_server.try_handle_info/3
    (stdlib 5.0.2) gen_server.erl:1165: :gen_server.handle_msg/6
    (stdlib 5.0.2) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
Last message: {:DOWN, #Reference<0.957762495.3125542913.23919>, :process, #PID<0.121572.0>, :normal}
```

**SDK version**

```shell
* ldclient 2.1.2 (Hex package) (rebar3)
  locked at 2.1.2 (launchdarkly_server_sdk) 78ce1735
```

**Language version, developer tools**

`elixir 1.14.5 (compiled with Erlang/OTP 26)`

**OS/platform**

`linux 5.10.186`

**Comment by kinyoklion** [comment]
Hello @zbarnes757,

Thank you for the issue. I will look into it.

This is happening immediately, or after some period of time?

_Filed internally as 217362_

Thank you,
Ryan

**Comment by zbarnes757** [comment]
It appears to happen after a period of time. I tend to get one of these errors per day on our QA instance.

**Comment by kinyoklion** [comment]
@zbarnes757 Are you setting `stream_initial_retry_delay_ms` when starting the client? This same error would occur if that value was set to something invalid. (The backoff would be triggered once a connection was lost at some point in the future).

Thank you,
Ryan

**Comment by kinyoklion** [comment]
@zbarnes757 You should not have to it should default to 1000, but I wanted to check as that field as it is the only way I have been able to produce a similar result. I will continue to look into it.

Thank you,
Ryan

**Comment by kinyoklion** [comment]
@zbarnes757 It looks like this would also happen when attempting to reconnect to launchdarkly 1015 times in a row and not succeeding. (which with the logic for backoff would be somewhere between 4-8 hours of reconnect attempts.)

Does this only happen on your QA instances?

**Comment by zbarnes757** [comment]
That is correct. Just in the QA instance. Everything else it working fine.

**Comment by kinyoklion** [comment]
@zbarnes757 I have released 3.0.1 that should resolve this problem for very large number of retries.

Thanks,
Ryan

**Comment by zbarnes757** [comment]
Thanks @kinyoklion! I've updated our dependency and have not seen any more errors yet. I'll post back here if it happens again.
